pkg/apis/internal.acorn.io/v1/appinstance.go

```diff
@@ -122,6 +123,18 @@ func (in *AppInstance) GetStopped() bool {
 	return in.Spec.Stop != nil && *in.Spec.Stop && in.DeletionTimestamp.IsZero()
 }
 
+// GetAllContainerNames returns a string slice containing the name of every container, job, and sidecar defined in Status.AppSpec.
+func (in *AppInstance) GetAllContainerNames() []string {
+	allContainers := append(maps.Keys(in.Status.AppSpec.Containers), maps.Keys(in.Status.AppSpec.Jobs)...)
+	for _, container := range in.Status.AppSpec.Containers {
+		allContainers = append(allContainers, maps.Keys(container.Sidecars)...)
+	}
+	for _, job := range in.Status.AppSpec.Jobs {
+		allContainers = append(allContainers, maps.Keys(job.Sidecars)...)
+	}
+	return allContainers
```

Comment:
nit: are these already sorted? if not, it would be nice to sort them

Reply:
Not sure, but I will go check

Reply:
They weren't sorted, so I pushed a change with code that sorts it before returning